src/arithmetic/string_funcs/string_funcs.c

```diff
@@ -222,9 +222,10 @@ SIValue AR_JOIN(SIValue *argv, int argc, void *private_data) {
 		delimiter = argv[1].stringval;
 	}
 
-	uint32_t count = SIArray_Length(list);
+	const uint32_t count = SIArray_Length(list);
+	const size_t delimeter_len = strlen(delimiter);
 
-	size_t delimeter_len = strlen(delimiter);
+	// Calculate the length of the resulting string.
 	uint str_len = delimeter_len * (count - 1);
```

Comment:
I believe the same overflow can happen where the delimiter is large and the number of elements in the array is significant. e.g. large delimiter * larger number of strings